integration-test/common/src/test/java/org/apache/uniffle/test/QuorumTest.java

```diff
@@ -427,18 +431,25 @@ public void case3() throws Exception {
     } catch (Exception e) {
       assertTrue(e.getMessage().startsWith("Get shuffle result is failed"));
     }
+    shuffleServers.set(1, createServer(1, tmpDir));
+    shuffleServers.get(1).start();
 
     // When the timeout of one server is recovered, the block sending should success
     disableTimeout((MockedShuffleServer)shuffleServers.get(2));
     report = shuffleWriteClientImpl.getShuffleResult("GRPC",
       Sets.newHashSet(shuffleServerInfo0, shuffleServerInfo1, shuffleServerInfo2),
       testAppId, 0, 0);
     assertEquals(report, blockIdBitmap);
+
   }
 
   @Test
   public void case4() throws Exception {
     String testAppId = "case4";
+    /** We must wait until server1 finish start, because {@link #case3} will restart server1. */
+    await().timeout(10, TimeUnit.SECONDS).until(
+        () -> isPortAvailable(SHUFFLE_SERVER_PORT + 1));
+    Thread.sleep(500);
```

Comment:
If i remove this line, this ut case will fail sometimes. i don't find the root cause.

Reply:
> If i remove this line, this ut case will fail sometimes. i don't find the root cause.
> https://github.com/xianjingfeng/incubator-uniffle/actions/runs/4460910585/jobs/7834366795

Seems it still fail sometimes.
https://github.com/kaijchen/incubator-uniffle/actions/runs/4464176186/jobs/7840089348

Reply:
I change it to sleep 1 second and run for 5 times. And it looks normal.
https://github.com/xianjingfeng/incubator-uniffle/actions/runs/4465348796
PS: some errors are caused by download dependency failures.

Reply:
I think I have solved this problem. https://github.com/apache/incubator-uniffle/pull/673/commits/3020eb05d91c71736272029f521d102777520183

https://github.com/xianjingfeng/incubator-uniffle/actions/runs/4485510017

By the way, i found that this method was added by you, I think you should be able to explain the underlying principle to me. @zuston 
https://github.com/apache/incubator-uniffle/blob/5fd68764d6fc9e085c89f9ddabd62d99eea6163e/integration-test/common/src/test/java/org/apache/uniffle/test/MultiStorageFaultToleranceBase.java#L56